### PR TITLE
add a 'watch' option to wire up a mutation observer

### PR DESCRIPTION
**Pull request by analogrelay**
The main change here is to add a `watch` option to `bootstrap`. When set to `true`, we hook up a `MutationObserver` to watch for changes to the DOM and automatically wire-up validation directives. This is essential if you have logic that is bringing new content into the DOM (such as when using a framework like [Turbo](http://turbo.hotwired.dev)).

I also added a little bit of opt-in logging (not much, but a start), and the ability to manually call `scan` on a given element to attach any validation directives.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -444,8 +463,14 @@ export class ValidationService {
     /**
      * Scans document for all validation message <span> generated by ASP.NET Core MVC, then tracks them.
      */
-    private scanMessages() {
-        let validationMessageElements = document.querySelectorAll('[data-valmsg-for]');
+    private scanMessages(root: HTMLElement) {
+        let validationMessageElements = Array.prototype.slice.call(root.querySelectorAll<HTMLElement>('[data-valmsg-for]')) as HTMLElement[];
```
Comment (dahlbyk) [inline src/index.ts:467]: Equivalent?
```suggestion
        let validationMessageElements = Array.from(root.querySelectorAll<HTMLElement>('[data-valmsg-for]'));
```

File: src/index.ts [hunk 2]
```diff
@@ -770,8 +795,14 @@ export class ValidationService {
     /**
      * Scans the entire document for input elements to be validated.
      */
-    private scanInputs() {
-        let inputs = document.querySelectorAll('[data-val="true"]');
+    private scanInputs(root: HTMLElement) {
+        let inputs = Array.prototype.slice.call(root.querySelectorAll<HTMLElement>('[data-val="true"]')) as HTMLElement[];
```
Comment (dahlbyk) [inline src/index.ts:799]:
```suggestion
        let inputs = Array.from(root.querySelectorAll<HTMLElement>('[data-val="true"]'));
```

**Review by dahlbyk — APPROVED** [review]

File: src/index.ts (hunk 2 above)
Comment (analogrelay) [reply src/index.ts:799]: Yeah, I tried this but TypeScript was unhappy. Perhaps the `tsconfig.json` is configured for an older set of APIs.

Comment (dahlbyk) [reply src/index.ts:799]: Seems like `NodeList` should be `ArrayLike` 🤷‍♂️ 

https://github.com/microsoft/TypeScript/blob/c6bef3f02874bddf6df120aa4f0d130c58478468/lib/lib.es2015.core.d.ts#L72
https://github.com/microsoft/TypeScript/blob/a7548745d78858f94f46f643b2a6edcec12e3a91/src/lib/es5.d.ts#L1503-L1506
https://github.com/microsoft/TypeScript/blob/c6bef3f02874bddf6df120aa4f0d130c58478468/lib/lib.dom.d.ts#L10069-L10097

Probably not worth a fight.

**Pull request merged**